src/KeyHandler.cpp

```diff
@@ -1301,39 +1378,52 @@ void KeyHandler::pinNode(
   }
 }
 
-void KeyHandler::pinNode(size_t cursor, const std::string& candidate,
-                         const std::string& associatePhrase) {
-  if (!grid_.overrideCandidate(cursor, candidate)) {
+void KeyHandler::pinNodeWithAssociatedPhrase(
+    size_t cursorIndex, const std::string& prefixReading,
+    const std::string& prefixValue, const std::string& associatedPhraseReading,
+    const std::string& associatedPhraseValue) {
+  // First, override the current node.
```

Comment:
I am using MS-Bopomofo style cursor, and it does not correctly insert the associated phrases. I guess we need to care about the line 1386 that sets cursor.

Reply:
Done.